src/data/energy.ts

```diff
@@ -432,6 +434,8 @@ const getEnergyData = async (
     }
   });
 
+  const statsMetadata = await getStatisticMetadata(hass, statIDs);
```

Comment:
Can we move some things around? We now fetch the metadata for gas statistics first and then fetch everything after. Can we just fetch all once?

Reply:
It's choice between two fetches or two loops over energy source as I have to write attr `source.unit_of_measurement` during first loop. I chose two fetches. I can filter out gasSources, so fetch would be smaller or make second loop with only gas sources.